auv_navigation/auv_navigation/src/navigation_utils/path_utils.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+
+import numpy as np
+import rospy
+from nav_msgs.msg import Path
+from geometry_msgs.msg import PoseStamped, TransformStamped
+import tf2_ros
+from tf.transformations import euler_from_quaternion, quaternion_from_euler, quaternion_matrix, quaternion_multiply, quaternion_inverse, quaternion_slerp
+from scipy.interpolate import CubicSpline
+import tf2_geometry_msgs
+
+
+
+def create_path_from_frame(tf_buffer, source_frame, target_frame, angle_offset=0.0, keep_orientation=False, num_samples=50):
+    try:
+        source_transform = tf_buffer.lookup_transform(
+            "odom", 
+            source_frame, 
+            rospy.Time(0),  
+            rospy.Duration(1.0)
+        )
+        target_transform = tf_buffer.lookup_transform(
+            "odom", 
+            target_frame, 
+            rospy.Time(0),  
+            rospy.Duration(1.0)
+        )
+        src_pos = source_transform.transform.translation
+        tgt_pos = target_transform.transform.translation
+        
+        src_q = [source_transform.transform.rotation.x,
+                source_transform.transform.rotation.y,
+                source_transform.transform.rotation.z,
+                source_transform.transform.rotation.w]
+        tgt_q = [target_transform.transform.rotation.x,
+                target_transform.transform.rotation.y,
+                target_transform.transform.rotation.z,
+                target_transform.transform.rotation.w]
+        
+        t_vals = np.linspace(0, 1, num_samples) 
+        
+        path = Path()
+        path.header.frame_id = "odom"
+        path.header.stamp = rospy.Time.now()
+        
+        # Apply angle offset
+        offset_q = quaternion_from_euler(0, 0, angle_offset)
+        final_q = quaternion_multiply(offset_q, tgt_q)
+        
+        for t in t_vals:
+            pose = PoseStamped()
+            pose.header = path.header
+            
+            pose.pose.position.x = src_pos.x + t * (tgt_pos.x - src_pos.x)
+            pose.pose.position.y = src_pos.y + t * (tgt_pos.y - src_pos.y)
+            pose.pose.position.z = src_pos.z + t * (tgt_pos.z - src_pos.z)
+            
+            # Set orientation based on keep_orientation flag
+            if keep_orientation:
+                interp_q = src_q
+            else:
+                interp_q = quaternion_slerp(src_q, final_q, t)
```

Comment:
We interpolate the yaw difference across the path. Any angle offset is added to the yaw difference.